library/aesce.c

```diff
@@ -65,6 +65,66 @@ int mbedtls_aesce_has_support(void)
 #endif
 }
 
+static uint8x16_t aesce_encrypt_block(uint8x16_t block,
+                                      unsigned char *keys,
+                                      int rounds)
+{
+    for (int i = 0; i < rounds - 1; i++) {
+        block = vaeseq_u8(block, vld1q_u8(keys + i * 16));
+        /* AES mix columns */
+        block = vaesmcq_u8(block);
+    }
+
+    /* AES single round encryption */
+    block = vaeseq_u8(block, vld1q_u8(keys + (rounds -1) * 16));
+
+    /* Final Add (bitwise Xor) */
+    block = veorq_u8(block, vld1q_u8(keys + rounds  * 16));
+
+    return block;
+}
+
+static uint8x16_t aesce_decrypt_block(uint8x16_t block,
+                                      unsigned char *keys,
+                                      int rounds)
+{
+
+    for (int i = 0; i < rounds - 1; i++) {
+        block = vaesdq_u8(block, vld1q_u8(keys + i * 16));
+        /* AES inverse mix columns */
```

Comment:
We are not following the AES algorithm as it is defined, this supposed to explain, how the implementation differs from the AES algorithm (and explains why this change is needed and why it is equivalent with the original).

As far as I can tell the [AESE documentation](https://developer.arm.com/documentation/ddi0597/2022-12/SIMD-FP-Instructions/AESE--AES-single-round-encryption-?lang=en) describes what `vaesdq_u8` does, but not the consequences on the implementation.

I think this comment adds value, because this wasn't immediately clear to me when I was first reading the code and took some time to figure out what is going on. I hope that this comment would save that time for other readers as well.